plugins/inputs/logfile/globpath/globpath.go

```diff
@@ -62,19 +62,6 @@ func (g *GlobPath) Match() map[string]os.FileInfo {
 		}
 		return out
 	}
-	if !g.hasSuperMeta {
-		out := make(map[string]os.FileInfo)
-		files, _ := filepath.Glob(g.path)
-		for _, file := range files {
-			info, err := os.Stat(file)
-			if info != nil {
-				out[file] = info
-			} else {
-				log.Printf("D! Stat file %v failed due to %v", g.path, err)
-			}
-		}
-		return out
-	}
```

Comment:
Why is it safe to remove this code block?

Does is duplicate functionality that is also in `walkFilePath()`?

Reply:
Because `filepath.Glob` doesn't behave how we want it to, and because the `walkFilePath` function does the same thing, but with the `glob.Glob` pattern

Reply:
Basically, the `TestInversionGlobPattern` test that I added is the important test that illustrates the need for this code change. If I put this block of code back in, the agent will call `filepath.Glob`, which will match `log2.log`, when what we really want is `log1.log`.

Reply:
If we're removing this block, then what's the point of the `hasSuperMeta` flag? We were using the `filepath.Glob` by default because it works for most situations and is more efficient.

Reply:
I can rip out the `hasSuperMeta` flag. I was just trying to be minimally disruptive. 

I guess now we come to a product decision. Which should we use? `filepath.Glob` is more efficient, but there is going to be confusion on why the agent doesn't match log files as customer expect with some glob patterns. 

As far as performance implications, I'll run something simple like `/tmp/*.log` and compare CPU and mem before and after the change since it would be comparing `filepath.Glob` and `Glob.Match`

Reply:
`hasSuperMeta` is basically the flag we use to tell the globpath to use Glob.Match. Right now it's only true if the path contains `**`. We can change it so it's also true if we need to handle unsupported glob characters.

Reply:
<img width="1227" alt="Screenshot 2023-08-15 at 4 22 31 PM" src="https://github.com/aws/amazon-cloudwatch-agent/assets/17056456/f2a4631a-d877-4737-9652-4ca45518e50a">

This is me running through the repro case, which isn't super useful from a performance standpoint, but interesting to see the weird difference. 

Making some adjustments then I'll run through a simple pattern (not the repro case) to compare performance